src/components/utils/BackButton.js

```diff
@@ -8,39 +8,75 @@ import React from "react";
 import { useTranslation } from "i18n";
-
+import ConfirmationDialog from "components/utils/ConfirmationDialog";
 import { Button, IconButton, useMediaQuery, useTheme } from "@material-ui/core";
 
 import { ArrowBack } from "@material-ui/icons";
 
 export default function BackButton(props) {
+    const { dirty = false } = props;
     const router = useRouter();
     const theme = useTheme();
     const isMobile = useMediaQuery(theme.breakpoints.down("xs"));
     const { t } = useTranslation("common");
 
+    const [showConfirmationDialog, setShowConfirmationDialog] =
+        React.useState(false);
+
     const onClick = () => router && router.back();
 
     return isMobile ? (
-        <IconButton
-            color="primary"
-            edge="start"
-            aria-label={t("back")}
-            onClick={onClick}
-            {...props}
-        >
-            <ArrowBack />
-        </IconButton>
+        <>
+            <IconButton
+                color="primary"
+                edge="start"
+                aria-label={t("back")}
+                onClick={() => {
+                    if (dirty) {
+                        setShowConfirmationDialog(true);
+                    } else {
+                        onClick();
+                    }
+                }}
+                {...props}
+            >
+                <ArrowBack />
+            </IconButton>
+            <ConfirmationDialog
+                open={showConfirmationDialog}
+                title={t("confirmDiscardChangesDialogHeader")}
+                contentText={t("confirmDiscardChangesDialogMsg")}
+                confirmButtonText={t("common:discard")}
+                onConfirm={onClick}
+                onClose={() => setShowConfirmationDialog(false)}
+            />
+        </>
     ) : (
-        <Button
-            color="primary"
-            variant={"contained"}
-            size="small"
-            startIcon={<ArrowBack fontSize="small" />}
-            onClick={onClick}
-            {...props}
-        >
-            {t("back")}
-        </Button>
+        <>
+            <Button
+                color="primary"
+                variant={"contained"}
+                size="small"
+                startIcon={<ArrowBack fontSize="small" />}
+                onClick={() => {
+                    if (dirty) {
+                        setShowConfirmationDialog(true);
+                    } else {
+                        onClick();
+                    }
+                }}
+                {...props}
+            >
+                {t("back")}
+            </Button>
+            <ConfirmationDialog
+                open={showConfirmationDialog}
+                title={t("confirmDiscardChangesDialogHeader")}
+                contentText={t("confirmDiscardChangesDialogMsg")}
+                confirmButtonText={t("common:discard")}
+                onConfirm={onClick}
+                onClose={() => setShowConfirmationDialog(false)}
+            />
```

Comment:
Minor, but this render function could be refactored so the confirmation dialog's code isn't in here twice